### PR TITLE
Move shared folder from devcontainer-setup

### PR DESCRIPTION
**Pull request by dennismeister93**
This PRs moves the content of the _shared folder from devenv-devcontainer-setup into velocitas-lib

### CODE REVIEW

File: velocitas_lib/conan_helper.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+def move_generated_sources(
+    generated_source_dir: str, output_dir: str, include_dir_rel: str, src_dir_rel: str
+) -> Tuple[List[str], List[str]]:
+    """Move generated source code from the generation dir into
+    headers: <output_dir>/<include_dir_rel>
+    sources: <output_dir>/<src_dir_rel>
+
+    Args:
+        generated_source_dir (str): The directory containing the generated sources.
```
Comment (MP91) [inline velocitas_lib/conan_helper.py:49]: We should generalize this a bit. Basically this method is just taking an input directory with header (*.h) and sources (*.cc) and puts these files into different folders. It doesn't really matter if the files are generated or not.

Comment (dennismeister93) [reply velocitas_lib/conan_helper.py:49]: give a proposal please, i just copied the content

Comment (doosuu) [reply velocitas_lib/conan_helper.py:49]: Actually, this method is very specific to the gRPC generator and probably should not even be moved into velocitas-lib

File: velocitas_lib/conan_helper.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        headers_relative.append(os.path.join(include_dir_rel, rel_path))
+
+    sources_relative = []
+    for source in sources:
+        rel_path = os.path.relpath(source, generated_source_dir)
+        os.makedirs(os.path.join(output_dir, src_dir_rel), exist_ok=True)
+        shutil.move(source, os.path.join(output_dir, src_dir_rel, rel_path))
+        sources_relative.append(os.path.join(src_dir_rel, rel_path))
+
+    return headers_relative, sources_relative
```
Comment (MP91) [inline velocitas_lib/conan_helper.py:77]:
```suggestion
def move_sources(
    source_dir: str, output_dir: str, include_dir_rel: str, src_dir_rel: str
) -> Tuple[List[str], List[str]]:
    """Move source and header files from the source dir into
    headers: <output_dir>/<include_dir_rel>
    sources: <output_dir>/<src_dir_rel>

    Args:
        source_dir (str): The directory containing the source (*.cc) and header (*.h) files.
        output_dir (str): The root directory to move the input files to.
        include_dir_rel (str): Path relative to output_dir where to move the headers to.
        src_dir_rel (str): Path relative to the output_dir where to move the sources to.

    Returns:
        Tuple[List[str], List[str]]: A tuple containing
            [0] = a list of the paths to all headers
            [1] = a list of the paths to all sources
    """

    headers = glob.glob(os.path.join(source_dir, "*.h"))
    sources = glob.glob(os.path.join(source_dir, "*.cc"))

    headers_relative = []
    for header in headers:
        rel_path = os.path.relpath(header, source_dir)
        os.makedirs(os.path.join(output_dir, include_dir_rel), exist_ok=True)
        shutil.move(header, os.path.join(output_dir, include_dir_rel, rel_path))
        headers_relative.append(os.path.join(include_dir_rel, rel_path))

    sources_relative = []
    for source in sources:
        rel_path = os.path.relpath(source, source_dir)
        os.makedirs(os.path.join(output_dir, src_dir_rel), exist_ok=True)
        shutil.move(source, os.path.join(output_dir, src_dir_rel, rel_path))
        sources_relative.append(os.path.join(src_dir_rel, rel_path))

    return headers_relative, sources_relative
```

File: velocitas_lib/conan_helper.py (hunk 1 above)
Comment (MP91) [reply velocitas_lib/conan_helper.py:49]: I would disagree, as mentioned basically it is just splitting the sources from the headers and puts them into new folders. Totally fine to have this in the velocitas_lib, IMHO

Comment (dennismeister93) [reply velocitas_lib/conan_helper.py]: to move or no to move. this is the question

**Review by doosuu — APPROVED** [review]
lgtm

**Pull request merged**